src/App.tsx

```diff
@@ -15,36 +16,102 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+interface SortOptions {
+  sortField: 'alphabetically' | 'length' | null;
+  isReverse: boolean;
+}
+
+function getPreparedList(
+  goods: string[],
+  { sortField, isReverse }: SortOptions,
+): string[] {
+  const list = [...goods];
+
+  if (sortField) {
+    list.sort((good1, good2) => {
+      switch (sortField) {
+        case 'alphabetically':
+          return good1.localeCompare(good2);
+        case 'length':
+          return good1.length - good2.length;
+
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    return list.reverse();
+  }
+
+  return list;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortOptions['sortField']>(null);
+  const [isReverse, setReverse] = useState(false);
+
+  const visibleGoods = getPreparedList(goodsFromServer, {
+    sortField,
+    isReverse,
+  });
+
+  function reset(): void {
+    setSortField(null);
+    setReverse(false);
+  }
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={classNames('button', 'is-info', {
+            'is-light': sortField !== 'alphabetically',
+          })}
```

Comment:
Consider simplifying the `classNames` usage by directly using a template string or conditional class application. This can improve readability and maintainability of the code.